x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/ExpressionBuilder.java

```diff
@@ -270,12 +270,6 @@ public LikePattern visitPattern(PatternContext ctx) {
         if (pattern == null) {
             throw new ParsingException(source(ctx.value), "Pattern must not be [null]");
         }
-        int pos = pattern.indexOf('*');
-        if (pos >= 0) {
-            throw new ParsingException(source(ctx.value),
-                    "Invalid char [*] found in pattern [{}] at position {}; use [%] or [_] instead",
-                    pattern, pos);
-        }
```

Comment:
@costin Did I miss anything, is there a reason to keep this check here?

Reply:
To prevent folks from using `*` instead of `%`. Whether that's still a concern I don't know.

Reply:
We don't have anything in the docs regarding the `*` and we list only the `%` and `_` 
`*` should be interpreted as a normal char for SQL:
```
postgres=# create table mytable (str varchar(10));
CREATE TABLE

postgres=# insert into mytable values('*test*'), ('as*test*fd');
INSERT 0 2

postgres=# select * from mytable where str like ('%*test*%');
    str
------------
 *test*
 as*test*fd
(2 rows)

postgres=# select * from mytable where str like ('*test*');
  str
--------
 *test*
(1 row)
```

How about adding a NOTE in the docs to explicitly specify that `*`, `?` and the usual regex chars are interpreted as normal chars?

Reply:
AFAIK it is only the MS Access SQL flavour that treats `*` as a wildcard in a `LIKE` pattern, so SQL folks should be fairly used to `*` not being a wildcard. ES customers with less SQL knowledge might not. I think adding a NOTE to the docs does not hurt.

Reply:
I am also thinking about removing the restriction on not being able to use `*` as an `ESCAPE` character from (here)[https://github.com/elastic/elasticsearch/blob/50fc9d52097cab6ac8b55625fc8b8f387688de13/x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/ExpressionBuilder.java#L285]. Do you see any good reason to keep that restriction?
As a user `*` would not be my first choice as an `ESCAPE` character, but the parser can handle it since `*` is just a normal character.

Reply:
On PostgreSQL, MySQL, SQLite any character can be used as ESCAPE (even wildchars) and any character can be escaped, not just wildcards. PostgreSQL treats the `'%%' ESCAPE '%'` as a non-wildcard `%` character (just as another escaped character), while MySQL and SQLite treat it as a wildcard (special escaping case). I assume it was intentional to behave differently and only allow escaping of the wildchars.

Reply:
As I see it, since only `%` and `_` are wildcards for `LIKE` I would only allow escaping of those, but if we go one with this change it shouldn't be part of the bugfix that would be backported, but for next minor release since it's a minor "breaking" change.

Reply:
Agree @matriv that allowing escaping any characters should be a separate PR and likely should be in the next minor release only, although technically it is less of a breaking change unless someone depends on the exceptions around escaping non-wildchars. Anyways, this is off topic for this PR.

To clarify: in this current PR I am only removing the special treatment of `*` and the restriction on using `*`as an `ESCAPE` character (wildchars are still not allowed as `ESCAPE` characters). Does this sound good to you?

Reply:
Sorry for the confusion, but if `*` is not a wildcard anymore, maybe worths removing it as escape character as well in this fix? @costin what do you think?